### PR TITLE
adjust oom_score of containerd-shim of watchdog

### PR DESCRIPTION
**Pull request by giggsoff**
In case of no hardware watchdog we are forced to use software one as a process with containerd.
We set oom_score_adj for workload (init.sh and child processes), but seems we do not set oom_score_adj for containerd-shim. 
```
7601d48f-084f-46e8-b7c2-8220b2c7cd6e:~# ps aux|grep watchdog
 1400 root      0:00 /usr/bin/containerd-shim-runc-v2 -namespace services.linuxkit -id watchdog -address /run/containerd/containerd.sock
 2071 root      0:00 /usr/sbin/watchdog -F
7601d48f-084f-46e8-b7c2-8220b2c7cd6e:~# cat /proc/2881/oom_score_adj 
-1000
7601d48f-084f-46e8-b7c2-8220b2c7cd6e:~# cat /proc/1400/oom_score_adj 
1
```
In this case with OOM in containerd cgroup we will hit `cleaning up dead shim` and software watchdog will be killed.

In this PR I set `oom_score_adj` for parent of `init.sh` of watchdog (it is containerd-shim) to be equal or lower than init.sh`s one to reduce probability of killing containerd-shim of watchdog.

The second commit is about adding of missed wdctl into watchdog

Signed-off-by: Petr Fedchenkov <giggsoff@gmail.com>

### CODE REVIEW

File: pkg/watchdog/init.sh [hunk 1]
```diff
@@ -64,6 +75,8 @@ if [ -c /dev/watchdog ]; then
 else
     log "Platform has no /dev/watchdog"
     USE_HW_WATCHDOG=0
+    # in case of no hardware watchdog we must survive OOM for watchdog`s parent process, i.e. containerd-shim
+    adjust_parent_oom_score
```
Comment (eriknordmark) [inline pkg/watchdog/init.sh:79]: Sure, but wouldn't we want to do the adjustment even if we have hardware watchdog? Makes for less different cases to think about if all of the processes for the watchdog container have the same score.

File: pkg/watchdog/init.sh [hunk 2]
```diff
@@ -4,6 +4,17 @@ USE_HW_WATCHDOG=1
 DEFAULT_WATCHDOG_CHANGE_TIME=300
 WATCHDOG_CHANGE_TIME=$(</proc/cmdline grep -o '\bchange=[^ ]*' | cut -d = -f 2)
 
+adjust_parent_oom_score() {
+  SELF_OOM_SCORE_ADJ=$(cat /proc/$$/oom_score_adj)
+  PARENT_OOM_SCORE_ADJ=$SELF_OOM_SCORE_ADJ
+  if [ -n "$SELF_OOM_SCORE_ADJ" ] && [ "$SELF_OOM_SCORE_ADJ" -eq "$SELF_OOM_SCORE_ADJ" ] 2>/dev/null; then
```
Comment (eriknordmark) [inline pkg/watchdog/init.sh:10]: What is the eq check? That it is a number? Could you get something other than a number (or empty string) from the cat of that file?

File: pkg/watchdog/init.sh [hunk 3]
```diff
@@ -4,6 +4,17 @@ USE_HW_WATCHDOG=1
...
+    if [ "$SELF_OOM_SCORE_ADJ" -gt "-1000" ]; then
+        PARENT_OOM_SCORE_ADJ=$(( SELF_OOM_SCORE_ADJ-1 ))
+    fi
```
Comment (eriknordmark) [inline pkg/watchdog/init.sh:13]: Why no always set it to the same? Isn't that sufficient?

File: pkg/watchdog/init.sh (hunk 1 above)
Comment (giggsoff) [reply pkg/watchdog/init.sh:79]: Why not, will do

File: pkg/watchdog/init.sh (hunk 2 above)
Comment (giggsoff) [reply pkg/watchdog/init.sh:10]: Good point, looks like noop

**Review by eriknordmark — COMMENTED** [review]
What's the motivation for adding wdctl? For debugging by EVE developers or something else?
FWIW wdctl is in both the debug and pillar containers.

**Comment by giggsoff** [comment]
> What's the motivation for adding wdctl? For debugging by EVE developers or something else?
> FWIW wdctl is in both the debug and pillar containers.

I can see it inside `init.sh` of watchdog, but we do not have it inside.

**Comment by eriknordmark** [comment]
> > What's the motivation for adding wdctl? For debugging by EVE developers or something else?
> > FWIW wdctl is in both the debug and pillar containers.
> 
> I can see it inside `init.sh` of watchdog, but we do not have it inside.

Odd - I guess the disable setting hasn't worked for a while.

**Review by eriknordmark — APPROVED** [review]
LGTM

**Review by rvs — COMMENTED** [review]
LGTM

**Pull request merged**